### PR TITLE
Feat/algo constraints

### PR DESCRIPTION
**Pull request by marcoviappiani**
#45
- Refactored algorithm to enforce constraints
- Updated README to indicate Angular instead of Backbone and updated project name to Seatly
- tested for multiple scenarios, still passes previous test cases.

current behaviour in edge cases (here our algorithm could be improved with future releases):
- if table has odd number of seats and there are more couples than we can fit, it's going to exceed number of seats per table
- if it doesn't find a solution immediately, it's going to drop constraints and not test a different sorting
- it should surface errors to the user (now it's logging to the console)
- it's always going to sort in the same way (there is no shuffling of seats)

### CODE REVIEW

File: server/utils/diningTableAlgo.js [hunk 1]
```diff
@@ -11,75 +20,202 @@ exports.makeDiningTables = function(guests, numPerTable) {
     }
   };
 
-  // define number of tables
-  var numTables = Math.ceil(guests.length / numPerTable);
+  var canAddFriend = function(diningTable, friend1, friend2) {
+    var result = true;
```
Comment (batmanimal) [inline server/utils/diningTableAlgo.js:25]: Would be helpful to add some comments in this function to explain in plain english what it is used for

File: server/utils/diningTableAlgo.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
 // input: an array of guest objects, max number of guests per table
 // only dealing with +1's for now
```
Comment (batmanimal) [inline server/utils/diningTableAlgo.js:3]: TODO: make sure to declare the output, not just the input up top here. i.e. "The output is an array of arrays, where the sub-arrays represent dining tables, populated by elements that are guest objects."

**Comment by batmanimal** [comment]
Please add comments and resubmit! Thanks.
